### ISSUE TITLE
ABIDE result question

### ISSUE THREAD

**Issue by Roninddd** [NZWANG/MM-GTUNets#6]
Hello, could you please tell me why the test accuracy of my ABIDE dataset is only around 0.5?

Start pretraining vae...
Epoch: 0,	lr: 0.00100,	loss: 361922166784.00000
Epoch: 100,	lr: 0.00100,	loss: 489589.09375
Epoch: 200,	lr: 0.00100,	loss: 474761.84375
Epoch: 300,	lr: 0.00100,	loss: 473534.37500
Epoch: 400,	lr: 0.00100,	loss: 473184.21875
Epoch: 500,	lr: 0.00100,	loss: 473069.46875
Epoch: 600,	lr: 0.00100,	loss: 473022.59375
Epoch: 700,	lr: 0.00100,	loss: 473001.03125
Epoch: 800,	lr: 0.00100,	loss: 472989.06250
Epoch: 900,	lr: 0.00100,	loss: 472990.09375
Epoch: 1000,	lr: 0.00100,	loss: 472978.18750
Epoch: 1100,	lr: 0.00100,	loss: 472979.62500
Epoch: 1200,	lr: 0.00100,	loss: 472979.03125
Early stopping

 => Fold 9 best pretrain vae loss 472969.53125, epoch 1175
=> Average training time in 10-fold CV: 59.56165 s
=> Average test accuracy in 10-fold CV: 0.49999(0.0005)
=> Average test sensitivity in 10-fold CV: 0.52226(0.2017)
=> Average test specificity in 10-fold CV: 0.47407(0.1890)
=> Average test AUC in 10-fold CV: 0.47804(0.0027)
=> Average test F1-score in 10-fold CV: 0.39064(0.0933)

**Comment by clh0x123** [comment]
Please provide detailed parameter settings and device operation configurations. Based on the information provided, we are unable to identify the reasons why the experimental results cannot be reproduced.

**Comment by Roninddd** [comment]
Could you please help me check if there is something wrong with the parameters?




==========&nbsp; &nbsp; &nbsp; &nbsp;CONFIG&nbsp; &nbsp; &nbsp; =============
data_folder:data/ABIDE_aal
atlas:aal
num_rois:116
num_classes:2
num_subjects:871
key:SUB_ID
labels:DX_GROUP
ages:AGE_AT_SCAN
genders:SEX
sites:SITE_ID
scores:['SITE_ID', 'SEX', 'AGE_AT_SCAN']
variable:connectivity
dataset:ABIDE
model:MM_GTUNets
node_dim:500
img_depth:2
ph_depth:3
hidden:128
out:16
dropout:0.3
edge_drop:0.3
pool_ratios:0.8
smh:1
deg:0.0001
val:0.01
use_cpu:False
gpu_id:0
train:True
seed:911
early_stop:100
lr:0.0001
vae_lr:0.001
wd:0.0005
epoch:500
folds:10
log_save:False
model_save:True
result_save:True
print_freq:5
ckpt_path:./save_model/MM_GTUNets_ABIDE_aal/
time:250416
device:cuda:0
==========&nbsp; &nbsp; &nbsp;CONFIG END&nbsp; &nbsp; =============





Existence
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "NZWANG/MM-GTUNets"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年4月16日(星期三) 中午12:10
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [NZWANG/MM-GTUNets] ABIDE result question (Issue #6)



   
Please provide detailed parameter settings and device operation configurations. Based on the information provided, we are unable to identify the reasons why the experimental results cannot be reproduced.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  clh0x123 left a comment (NZWANG/MM-GTUNets#6)
 
Please provide detailed parameter settings and device operation configurations. Based on the information provided, we are unable to identify the reasons why the experimental results cannot be reproduced.

**Comment by clh0x123** [comment]
The experimental parameters you provided are correct, but the model's loss function fails to converge on your device. I believe this issue may stem from the deployment environment or hardware configuration. It is recommended that this project be run on a system with at least an NVIDIA GeForce 2080 Ti or higher. Please also ensure that your PyTorch environment is correctly installed with the GPU-supported version. Below are the results obtained in my configured environment:
=> Average training time in 10-fold CV: 192.45122 s
=> Average test accuracy in 10-fold CV: 0.82104(0.0031)
=> Average test sensitivity in 10-fold CV: 0.85278(0.0047)
=> Average test specificity in 10-fold CV: 0.78427(0.0053)
=> Average test AUC in 10-fold CV: 0.88448(0.0035)
=> Average test F1-score in 10-fold CV: 0.83621(0.0027)
✔ Saved model to:./save_model/MM_GTUNets_ABIDE_aal/

**Comment by Roninddd** [comment]
Thank you very much for your reply and guidance.




Existence
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "NZWANG/MM-GTUNets"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年4月16日(星期三) 下午3:22
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [NZWANG/MM-GTUNets] ABIDE result question (Issue #6)



   
The experimental parameters you provided are correct, but the model's loss function fails to converge on your device. I believe this issue may stem from the deployment environment or hardware configuration. It is recommended that this project be run on a system with at least an NVIDIA GeForce 2080 Ti or higher. Please also ensure that your PyTorch environment is correctly installed with the GPU-supported version. Below are the results obtained in my configured environment:
 =&gt; Average training time in 10-fold CV: 192.45122 s
 =&gt; Average test accuracy in 10-fold CV: 0.82104(0.0031)
 =&gt; Average test sensitivity in 10-fold CV: 0.85278(0.0047)
 =&gt; Average test specificity in 10-fold CV: 0.78427(0.0053)
 =&gt; Average test AUC in 10-fold CV: 0.88448(0.0035)
 =&gt; Average test F1-score in 10-fold CV: 0.83621(0.0027)
 ✔ Saved model to:./save_model/MM_GTUNets_ABIDE_aal/

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  clh0x123 left a comment (NZWANG/MM-GTUNets#6)
 
The experimental parameters you provided are correct, but the model's loss function fails to converge on your device. I believe this issue may stem from the deployment environment or hardware configuration. It is recommended that this project be run on a system with at least an NVIDIA GeForce 2080 Ti or higher. Please also ensure that your PyTorch environment is correctly installed with the GPU-supported version. Below are the results obtained in my configured environment:
 =&gt; Average training time in 10-fold CV: 192.45122 s
 =&gt; Average test accuracy in 10-fold CV: 0.82104(0.0031)
 =&gt; Average test sensitivity in 10-fold CV: 0.85278(0.0047)
 =&gt; Average test specificity in 10-fold CV: 0.78427(0.0053)
 =&gt; Average test AUC in 10-fold CV: 0.88448(0.0035)
 =&gt; Average test F1-score in 10-fold CV: 0.83621(0.0027)
 ✔ Saved model to:./save_model/MM_GTUNets_ABIDE_aal/

**Comment by Roninddd** [comment]
Hello, I think there might be some issues with my processing of the ABIDE dataset. If it's not too much trouble, could you please provide me with the ABIDE aal folder? Thank you.




---Original---
From: ***@***.***&gt;
Date: Wed, Apr 16, 2025 15:23 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NZWANG/MM-GTUNets] ABIDE result question (Issue #6)


   
The experimental parameters you provided are correct, but the model's loss function fails to converge on your device. I believe this issue may stem from the deployment environment or hardware configuration. It is recommended that this project be run on a system with at least an NVIDIA GeForce 2080 Ti or higher. Please also ensure that your PyTorch environment is correctly installed with the GPU-supported version. Below are the results obtained in my configured environment:
 =&gt; Average training time in 10-fold CV: 192.45122 s
 =&gt; Average test accuracy in 10-fold CV: 0.82104(0.0031)
 =&gt; Average test sensitivity in 10-fold CV: 0.85278(0.0047)
 =&gt; Average test specificity in 10-fold CV: 0.78427(0.0053)
 =&gt; Average test AUC in 10-fold CV: 0.88448(0.0035)
 =&gt; Average test F1-score in 10-fold CV: 0.83621(0.0027)
 ✔ Saved model to:./save_model/MM_GTUNets_ABIDE_aal/

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  clh0x123 left a comment (NZWANG/MM-GTUNets#6)
 
The experimental parameters you provided are correct, but the model's loss function fails to converge on your device. I believe this issue may stem from the deployment environment or hardware configuration. It is recommended that this project be run on a system with at least an NVIDIA GeForce 2080 Ti or higher. Please also ensure that your PyTorch environment is correctly installed with the GPU-supported version. Below are the results obtained in my configured environment:
 =&gt; Average training time in 10-fold CV: 192.45122 s
 =&gt; Average test accuracy in 10-fold CV: 0.82104(0.0031)
 =&gt; Average test sensitivity in 10-fold CV: 0.85278(0.0047)
 =&gt; Average test specificity in 10-fold CV: 0.78427(0.0053)
 =&gt; Average test AUC in 10-fold CV: 0.88448(0.0035)
 =&gt; Average test F1-score in 10-fold CV: 0.83621(0.0027)
 ✔ Saved model to:./save_model/MM_GTUNets_ABIDE_aal/

**Comment by clh0x123** [comment]
To obtain the preprocessed ABIDE data, please run the following command in the CMD:
```shell
python fetch_abide.py
```
For more details, refer to the following link: [nilearn.datasets.fetch_abide_pcp - Nilearn](https://nilearn.github.io/stable/modules/generated/nilearn.datasets.fetch_abide_pcp.html)

**Comment by Roninddd** [comment]
Thank you for your reply.



---Original---
From: ***@***.***&gt;
Date: Thu, Apr 17, 2025 09:46 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NZWANG/MM-GTUNets] ABIDE result question (Issue #6)


   
To obtain the preprocessed ABIDE data, please run the following command in the CMD:
 python fetch_abide.py
 
For more details, refer to the following link: nilearn.datasets.fetch_abide_pcp - Nilearn

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  clh0x123 left a comment (NZWANG/MM-GTUNets#6)
 
To obtain the preprocessed ABIDE data, please run the following command in the CMD:
 python fetch_abide.py
 
For more details, refer to the following link: nilearn.datasets.fetch_abide_pcp - Nilearn

**Comment by Roninddd** [comment]
Hello, I used the ABIDE dataset downloaded and processed with your code for training, and the accuracy was only around 0.5. However, I achieved an accuracy of around 0.71 using the ABIDE dataset that I had previously processed myself. What is the reason for this?




Existence
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "NZWANG/MM-GTUNets"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年4月17日(星期四) 上午9:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [NZWANG/MM-GTUNets] ABIDE result question (Issue #6)



   
To obtain the preprocessed ABIDE data, please run the following command in the CMD:
 python fetch_abide.py
 
For more details, refer to the following link: nilearn.datasets.fetch_abide_pcp - Nilearn

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  clh0x123 left a comment (NZWANG/MM-GTUNets#6)
 
To obtain the preprocessed ABIDE data, please run the following command in the CMD:
 python fetch_abide.py
 
For more details, refer to the following link: nilearn.datasets.fetch_abide_pcp - Nilearn

**Comment by clh0x123** [comment]
I’ve never encountered this issue before. In my opinion, it’s very likely caused by an incorrect environment setup on your end. If you’re having trouble configuring a deep learning environment yourself, I would recommend using a cloud platform like **AutoDL**, where you can rent a server with the necessary environment pre-configured. This can save you time and help ensure the project runs smoothly.
